pkg/devspace/build/localregistry/local_registry.go

```diff
@@ -164,6 +167,32 @@ func (r *LocalRegistry) RewriteImageForBuilder(image string) (string, error) {
 }
 
 func (r *LocalRegistry) ensureNamespace(ctx devspacecontext.Context) error {
+	// If localregistry namespace is the same as devspace, we don't have
+	// anything to do.
+	if r.Namespace == ctx.KubeClient().Namespace() {
+		return nil
+	}
+
+	// List all namespaces, this will already return an error in case of
+	// user's permissions problems.
+	namespaces, err := ctx.KubeClient().
```

Comment:
Why list and not get?

Reply:
I thought about it, but I wanted to be sure that an `err` from the client was for permission related issues
Because `namespace XXX does not exist` is also an `err` in the get, and I thought pattern-matching err wouldn't be optimal

Reply:
There is a function that helps with the error detection. [Here’s an example usage](https://github.com/devspace-sh/devspace/blob/b13898a69ec1431f584b4a5e3d4e26c13b3e645b/cmd/cleanup/local_registry.go#L100).

Reply:
Oh Thanks will look into it!

Reply:
Done :) much better this way, thanks!